Wisetara/deprecate args active support  test case#assert nothing raised for pr

wisetara opened the pull request:
Using arguments (to specify an exception error) with `assert_nothing_raised` is not recommended, as discussed with Rafael França on https://github.com/rails/rails/pull/23773. 

A deprecation warning has been added to `ActiveSupport::TestCase#assert_nothing_raised(*args)`, and all tests have been updated to use `assert_nothing_raised` without arguments.

Commit:
remove args from assert_nothing_raised in tests

Commit:
update assert_nothing_raised in testing.md

Commit:
add deprecation warning to assert_nothing_raised and changelog entry

wisetara commented:
Well, there may be a bit too much warning! I see that now. But I don't believe sorting that out will fix the failure: 
TemplateDigestorTest#test_logging_of_missing_template [/home/travis/build/rails/rails/actionview/test/template/digestor_test.rb:109]:
Expected /Couldn't\ find\ template\ for\ digesting:\ messages\/something_missing/ to match "".
If someone looks at it, would you #1, agree that perhaps a documented deprecation warning above the method is best rather than an actual warning (or I could code it to only show when arguments are passed in, I suppose), and #2, any thoughts on that failure.

prathamesh-sonpatki commented:
r? @rafaelfranca

meinac commented on activesupport/lib/active_support/test_case.rb:
```diff
@@ -75,6 +75,11 @@ def test_order
     #     perform_service(param: 'no_exception')
     #   end
     def assert_nothing_raised(*args)
+      ActiveSupport::Deprecation.warn(<<-MSG.squish)
```
Maybe you have to check presence of arguments. Now it's giving deprecation error every time even if you don't send any parameter to it.

wisetara commented:
That's better! I hope this deprecation PR is a helpful, @rafaelfranca! Please let me know if you'd like any additional changes.

rafaelfranca commented on activesupport/CHANGELOG.md:
```diff
@@ -1,3 +1,13 @@
+*   Deprecate `(*args)` on `assert_nothing_raised`.
+
+    Using arguments (to specify an exception error) with
```
Thank you for mentioning me but we usually don't do that.

What do you think about?

```
Deprecate arguments on `assert_nothing_raised`.

`assert_nothing_raised` were not asserting the arguments that were being passed
(usually an specific exception class) since it only yields the block. To not confuse the
users that the arguments have any meaning they are being deprecated.
```

rafaelfranca commented on activesupport/lib/active_support/test_case.rb:
```diff
@@ -75,6 +75,9 @@ def test_order
     #     perform_service(param: 'no_exception')
     #   end
     def assert_nothing_raised(*args)
+      ActiveSupport::Deprecation.warn("Passing arguments to
```
```
if args.present?
  ActiveSupport::Deprecation.warn("Passing arguments to assert_nothing_raised is deprecated and will be removed in Rails 5.1.")
end
```

rafaelfranca commented:
Awesome work. I made some comment.

wisetara commented:
Thank you for the comments and feedback. I've incorporated your `if-else` refactor on testcase.rb, and I modified the CHANGELOG.md. I've rebased, and I hope this is good, but if you'd like any other changes, please let me know. It feels good to contribute!

rafaelfranca commented:
Thank you
